### PR TITLE
Checkout: Fix dependency warnings in composite-checkout package

### LINKED ISSUE #50647: Missing dependencies: ./packages/composite-checkout/package.json
### Undeclared dependencies:
```
  ./packages/composite-checkout/src/components/order-review-line-items.tsx:11:1: Undeclared dependency on src
  ./packages/composite-checkout/src/components/radio-button.tsx:10:1: Undeclared dependency on src
```

### Missing peer dependencies
```
  ./packages/composite-checkout/package.json:53:32: Unmet transitive peer dependency on enzyme@^3.11.0, via @automattic/calypso-build@^7.0.0
  ./packages/composite-checkout/package.json:53:32: Unmet transitive peer dependency on jest@>=26.4.0, via @automattic/calypso-build@^7.0.0
  ./packages/composite-checkout/package.json:53:32: Unmet transitive peer dependency on postcss@^8.2.6, via @automattic/calypso-build@^7.0.0
```

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This adjusts the dependencies of the `@automattic/composite-checkout` package to be more accurate.

Fixes https://github.com/Automattic/wp-calypso/issues/50647

#### Testing instructions

Verify that there are no build errors.

### CODE REVIEW

File: packages/composite-checkout/package.json [hunk 1]
```diff
@@ -48,13 +48,16 @@
...
-		"react-dom": "^16.12.0",
 		"@automattic/calypso-build": "^7.0.0",
 		"@automattic/calypso-polyfills": "^1.0.0",
 		"@testing-library/jest-dom": "^5.9.0",
 		"@testing-library/react": "^10.0.5",
+		"jest": ">=26.4.0",
 		"webpack": "^5.24.4"
 	},
+	"peerDependencies": {
+		"react": "^16.12.0",
```
Comment (scinos) [inline packages/composite-checkout/package.json:63]: These needs to be, at least, `devDependencies`, as they are used by `@testing-library/react`. They can be `peerDependencies` as well.

File: packages/composite-checkout/package.json [hunk 2]
```diff
@@ -48,13 +48,16 @@
 		"react-stripe-elements": "^4.0.2"
 	},
 	"devDependencies": {
-		"react": "^16.12.0",
-		"react-dom": "^16.12.0",
 		"@automattic/calypso-build": "^7.0.0",
 		"@automattic/calypso-polyfills": "^1.0.0",
 		"@testing-library/jest-dom": "^5.9.0",
 		"@testing-library/react": "^10.0.5",
+		"jest": ">=26.4.0",
```
Comment (scinos) [inline packages/composite-checkout/package.json]: Using `"jest": "^26.4.0"` is consistent with the rest of the repo and satisfies the required peer dependency.
```suggestion
		"jest": "^26.4.0",
```

**Comment by scinos** [comment]
I have pushed a commit with the changes I suggest in the comments above. It should fix all the dependency related errors

```
$ npx @yarnpkg/doctor packages/composite-checkout

➤ YN0000: ┌ /Users/sergio/src/automattic/wp-calypso/packages/composite-checkout/package.json
➤ YN0000: │ /Users/sergio/src/automattic/wp-calypso/packages/composite-checkout/src/components/radio-button.tsx:10:1: Undeclared dependency on src
➤ YN0000: └ Completed in 3s 437ms
```

**Comment by sirbrillig** [comment]
Thank you! That looks fine to me.

**Pull request merged**